cli/flox/src/commands/install.rs

```diff
@@ -222,6 +227,25 @@ impl Install {
 
         let mut environment = concrete_environment.into_dyn_environment();
 
+        // Get a list of the packages that this environment is already overriding via composition.
+        let lockfile_path = environment.lockfile_path(&flox)?;
+        let canonicalized_lockfile_path_res = CanonicalPath::new(lockfile_path);
+        let existing_composer_package_overrides = if let Ok(path) = canonicalized_lockfile_path_res
+        {
+            let lockfile_content =
+                std::fs::read_to_string(&path).context("Failed to read lockfile")?;
+            let lockfile =
+                Lockfile::from_str(&lockfile_content).context("Failed to parse lockfile")?;
+            let existing_composer_overrides = lockfile
+                .compose
+                .map(|c| c.warnings)
+                .map(|warnings| package_overrides_for_manifest_id(&warnings, COMPOSER_MANIFEST_ID))
+                .unwrap_or_default();
+            Ok::<_, anyhow::Error>(existing_composer_overrides)
+        } else {
+            Ok(vec![])
```

Comment:
nit: I'm not seeing why the `Ok()` wrapping is needed, can we drop that?

Reply:
Both sides of the if-statement need to return `Result`, and the upper branch is returning errors early if we fail to read the lockfile.